rs/consensus/utils/src/pool_reader.rs

```diff
@@ -791,19 +792,21 @@ pub mod test {
     // sorted in ascending order.
     fn test_get_by_height_range() {
         ic_test_utilities::artifact_pool_config::with_test_pool_config(|pool_config| {
-            let Dependencies { mut pool, .. } = dependencies(pool_config, 1);
-            // Let's generate 4 proposals so that 2 of them end up in the
-            // unvalidated pool and have the same height.
             let rounds = 30;
-            let replicas = 3;
+            let replicas = 10;
+            let f = 3;
+            let Dependencies { mut pool, .. } = dependencies(pool_config, replicas);
+
+            // Half the block proposals will be validated, the other half stay
+            // unvalidated.
```

Comment:
It took me a while why this is the case. Can we explain that `TestConsensusPool::advance_round` puts every other block in the validated section?

Reply:
done